src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1641,6 +1650,43 @@ internal override void CreateAsync()
 
             // Create reader thread and send first PSRP message.
             StartReaderThread(_stdOutReader);
+
+            // Monitor SSH process until connection is established.
+            _sshProcessTimer = new Timer(
```

Comment:
i noticed that this is not CLS-Compliant - is that ok?
https://docs.microsoft.com/en-us/dotnet/api/system.threading.timer.-ctor?view=net-5.0#System_Threading_Timer__ctor_System_Threading_TimerCallback_System_Object_System_UInt32_System_UInt32_

Reply:
Not sure why it isn't CLS compliant, but we have a lot of non-compliant code in PowerShell so I think it is Ok.